src/receiver/k8sobjectsreceiver/receiver.go

```diff
@@ -0,0 +1,491 @@
+	}
+
+	statusChanged := true
+	metadataChanged := true
+	specChanged := true
+
+	storage.mu.Lock()
+	oldHashes, exists := storage.Objects[key]
+	if exists {
+		metadataChanged = oldHashes.Metadata != hashes.Metadata
+		specChanged = oldHashes.Spec != hashes.Spec
+		statusChanged = oldHashes.Status != hashes.Status
+	}
+	storage.Objects[key] = *hashes
+	storage.mu.Unlock()
+
+	if !metadataChanged && !statusChanged && !specChanged {
+		return nil
+	}
+
+	logs, err := watchObjectsToLogData(event, observedAt, config, func(attrs pcommon.Map) {
+		attrs.PutBool("sw.metadata.changed", metadataChanged)
+		attrs.PutBool("sw.status.changed", statusChanged)
+		attrs.PutBool("sw.spec.changed", specChanged)
+	})
+	if err != nil {
+		return fmt.Errorf("error converting watch objects to log data %w", err)
+	} else {
+		obsCtx := kr.obsrecv.StartLogsOp(ctx)
+		err := kr.consumer.ConsumeLogs(obsCtx, logs)
+		kr.obsrecv.EndLogsOp(obsCtx, metadata.Type.String(), 1, err)
+	}
+
+	return nil
+}
+
+func (kr *k8sobjectsreceiver) updateStorage(ctx context.Context, storage *objectstorage) error {
+	if kr.storageClient == nil || storage == nil {
+		return nil
+	}
+
+	dataBytes, err := json.Marshal(storage.Objects)
+	if err != nil {
+		return err
+	}
+
+	return kr.storageClient.Set(ctx, storage.key, dataBytes)
+}
+
+func (kr *k8sobjectsreceiver) loadStorage(ctx context.Context, storage *objectstorage) error {
+	if kr.storageClient == nil || storage == nil {
+		return nil
+	}
+	// load existing data from storage
+	dataBytes, err := kr.storageClient.Get(ctx, storage.key)
+	if err != nil {
+		return err
+	}
+
+	if len(dataBytes) > 0 {
+		storage.mu.Lock()
+		err = json.Unmarshal(dataBytes, storage.Objects)
+		storage.mu.Unlock()
+
+		if err != nil {
+			return fmt.Errorf("failed to unmarshal stored data: %w", err)
+		}
+
+		kr.setting.Logger.Info("Data loaded from storage", zap.Any("key", storage.key))
+	}
+
+	return nil
+}
+
+func getObjectHashes(udata *unstructured.Unstructured) (*objecthashes, error) {
+	metadataHash, err := getHash(udata, "metadata")
+	if err != nil {
+		return nil, err
+	}
+	specHash, err := getHash(udata, "spec")
+	if err != nil {
+		return nil, err
+	}
+	statusHash, err := getHash(udata, "status")
+	if err != nil {
+		return nil, err
+	}
+
+	return &objecthashes{
+		Metadata: metadataHash,
+		Spec:     specHash,
+		Status:   statusHash,
+	}, nil
+}
+
+func getHash(udata *unstructured.Unstructured, fields ...string) (string, error) {
+	nested, found, err := unstructured.NestedFieldCopy(udata.Object, fields...)
+	if err != nil {
+		return "", err
+	}
+
+	if !found {
+		return "", nil
+	}
+
+	bytes, err := json.Marshal(nested)
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("%x", sha256.Sum256(bytes)), nil
+}
+
+// key for the storage
+func getKey(udata *unstructured.Unstructured) string {
+	return fmt.Sprintf("%s:%s#%s", udata.GetKind(), udata.GetNamespace(), udata.GetName())
+}
```

Comment:
there are new methods for calculating hashes from manifests and saving it to storage